### PR TITLE
Add a --dry-run to pipeline upload that dumps json

### PR DESCRIPTION
**Pull request by lox**
This is a follow on of https://github.com/buildkite/agent/pull/765. 

It adds a `buildkite-agent pipeline upload --dry-run` command that outputs the pretty-printed pipeline json rather than uploading it. This works locally with no configuration required.

### CODE REVIEW

File: clicommand/pipeline_upload.go [hunk 1]
```diff
@@ -174,6 +181,30 @@ var PipelineUploadCommand = cli.Command{
...
+			if err := enc.Encode(parsed); err != nil {
+				logger.Fatal("%#v", err)
+			}
+
+			os.Exit(0)
+		}
+
+		// Check we have a job id set if not in dry run
+		if cfg.Job == "" {
+			logger.Fatal("Missing job parameter. Usually this is set in the environment for a Buildkite job via BUILDKITE_JOB_ID.")
```
Comment (lox) [inline clicommand/pipeline_upload.go]: I'd love a way to make more informative error messages like this the standard vs the standard "required parameter" message.

**Comment by lox** [comment]
@toolmantim sure! Would be good to verify it works with bksrsly.

**Comment by toolmantim** [comment]
I've just done some standalone tests using various pipeline.yml files, and the env interpolation and JSON output seems to work really nicely! 🎉

I jumped into bksrsly to update it to so it works with it, but it's gonna take a bit more work than I thought, because sub-processes etc. I'll let you know when I've got something working 👌🏼

**Comment by jgonera** [comment]
@lox Does this validate the schema of the pipeline too? Will it fail on invalid pipelines?

**Pull request merged**